src/nickel.rs

```diff
@@ -141,4 +142,15 @@ impl Nickel {
 
         Server::new(self.middleware_stack).serve(addr);
     }
+    
+    pub fn listen_https<T: ToSocketAddrs + Display>(mut self, addr: T, cert: &Path, key: &Path) {
+        self.middleware_stack.add_middleware(middleware! {
+            (StatusCode::NotFound, "File Not Found")
+        });
+
+        println!("Listening on https://{}", addr);
+        println!("Ctrl-C to shutdown server");
+
+        Server::new(self.middleware_stack).serve_https(addr, cert, key);
```

Comment:
I think `Server::new` should take an enum describing if it's HTTP or HTTPS, and then `Nickel::listen_http` defined in terms of `Nickel::listen_https` or at least factoring out common code. I don't think we need to add additional api to `Server` since it's internal and therefore not a big deal to have to import an enum variant.